pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -223,17 +223,19 @@ public void cleanupLLC(final String realtimeTableName) {
     // If there are any completions in the pipeline we let them commit.
     Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
 
-    IdealState idealState = HelixHelper.getTableIdealState(_helixManager, realtimeTableName);
     final List<String> segmentsToRemove = new ArrayList<String>();
-    Set<String> allSegments = idealState.getPartitionSet();
-    int removeCount = 0;
-    for (String segmentName : allSegments) {
-      if (SegmentName.isLowLevelConsumerSegmentName(segmentName)) {
-        segmentsToRemove.add(segmentName);
-        removeCount++;
+    IdealState idealState = HelixHelper.getTableIdealState(_helixManager, realtimeTableName);
+    if (idealState != null) {
```

Comment:
NIT for easier reading. 
       if (idealState == null) {
             return; 
        }
        .....